packages/effect/src/internal/rateLimiter.ts

```diff
@@ -0,0 +1,74 @@
+// Direct port of https://github.com/svroonland/rezilience/blob/master/rezilience/shared/src/main/scala/nl/vroste/rezilience/RateLimiter.scala
+import type { Duration } from "effect"
+import { Chunk, Deferred, Effect, pipe, Queue, Ref, Stream } from "effect"
```

Comment:
in library imports should come from individual files as this would prevent tree shaking in bundlers like esbuild

Reply:
Done.